include/zio/zio_dev.h

```diff
@@ -260,19 +309,27 @@ typedef int (*zio_dev_attach_buf_t)(struct device *dev, struct zio_buf *buf);
  */
 typedef int (*zio_dev_detach_buf_t)(struct device *dev);
 
+/**
+ * @brief Function to set a device attribute
+ *
+ * @return -EINVAL if an invalid attribute id was given, 0 otherwise
+ */
+typedef int (*zio_dev_set_attr_t)(struct device *dev, const u16_t attr_idx, const struct zio_variant var);
```

Comment:
This seems like it'd be unnecessary given each attribute description has a get/set function pointer